File: Backend/services/ad/src/main/java/propertyrenting/ad/model/AdImage.java
package propertyrenting.ad.model;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Table(name = "ad_image")
@NoArgsConstructor
@Getter
public class AdImage {

    @Id
    @GeneratedValue
    private Long id;

    @Setter
    @Column
    private String name;

    @Setter
    @Column
    private String type;

    @Setter
    @Column(length = 3000)
    private byte[] picByte;

    @Setter
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "ad", referencedColumnName = "id")
    private Ad ad;

    public AdImage(String name, String type, byte[] picByte) {
        this.name = name;
        this.type = type;
        this.picByte = picByte;
    }
}

File: Frontend/property-renting-app/src/app/core/model/ad-image.ts
export class AdImage {
  public name: string;
  public type: string;
  public picByte: any;

  constructor(name: string, type: string, picByte: any) {
    this.name = name;
    this.type = type;
    this.picByte = picByte;
  }
}
